src/WorldMap.cpp

```diff
@@ -104,29 +92,12 @@ sf::IntRect WorldMap::TileSheet::getTileLocationByPosition(const sf::IntRect& po
 	return sf::IntRect(position.left * tileSize, position.top * tileSize, position.width * tileSize, position.height * tileSize);
 }
 
-//CollidableTileLayer
-void WorldMap::CollidableTileLayer::setTileMap(const std::vector<std::vector<int>>& tileMapData, const LevelDetails& levelDetails)
+void WorldMap::CollidableTileLayer::addCollidableTile(const sf::Vector2i & tilePosition, int tileSize)
 {
-	if (!m_tileMap.empty())
-	{
-		m_tileMap.clear();
-	}
-
-	for (int row = 0; row < levelDetails.m_mapSize.y; ++row)
-	{
-		for (int col = 0; col < levelDetails.m_mapSize.x; ++col)
-		{
-			const int tileID = tileMapData[col][row];
-			const int tileSize = levelDetails.m_tileSize;
-			if (tileID > 0)
-			{
-				m_tileMap.emplace_back(row * tileSize, col * tileSize, tileSize, tileSize);
-			}
-		}
-	}
+	m_tileMap.emplace_back(tilePosition.x, tilePosition.y - 32);
```

Comment:
Simplified the way how the collision layer is parsed. Now each collidable tile is passed by its position only.